stdlib/gc.mli

```diff
@@ -45,10 +45,20 @@ type stat =
 
     live_words : int;
     (** Number of words of live data in the major heap, including the header
-       words. *)
+       words.
+
+       Note that "live" words refers to every word in the major heap that isn't
+       currently known to be collectable, which includes words that haven't been
+       reachable by the program since the start of the previous gc cycle. It is
```

Comment:
```suggestion
       currently known to be collectable, which includes words that have become
       unreachable by the program after the start of the current gc cycle. It is
```

Reply:
My understanding is that, in the following scenario:

```
let a = [| ref 0 ]}
Gc.full_major ();
(* point 1 *)
a.(0) <- ref 1;
a.(0) <- ref 2;
(* point 2 *)
Gc.full_major ();
(* point 3 *)
Gc.full_major ()
(* point 4 *)
```

I would expect the `ref 1` to be darkened in the cycle started after at point 1, switch to white between point 2 and 3, and be reclaimed between 3 and 4. And if you measure live words at point 3, `ref 1` would be counted as live despite not being reachable at the start of the cycle.

Is that not correct?

Reply:
I ran my program above and my explanation that there are live values from the previous gc cycle is correct.

Or more precisely, I ran a fixed version of that program where `Gc.full_major ()` is replaced `Gc.major ()`, as I never realized that `Gc.full_major ()` finishes the current gc cycle and finishes a new one too. So I'll change the suggestion to call `Gc.full_major` twice, as that doesn't seem necessary.

Reply:
Yes you're right. But then it should be "since the start of the **current** GC cycle" rather than previous. Anything that is unreachable at the start of a given cycle will be reclaimed by this cycle.

Reply:
IIUC we agree on the behavior of the GC and are just debating the phrasing of the doc.

I say "since the start of the previous gc cycle" because some blocks that becomes unreachable right after the start previous gc cycle can be counted live in the current cycle. And no block that became unreachable sooner can be live.

Your statement is true, but less specific. That makes it seem like `Gc.major ()` can be used instead of `Gc.full_major ()`.

I reworded "since the start of the previous gc cycle" as "since **after** the start of previous gc cycle", to clarify that a block must become unreachable after the previous gc cycle started, not when the previous gc cycle started.